libraries/GCS_MAVLink/GCS_MAVLink.h

```diff
@@ -45,10 +45,14 @@ extern mavlink_system_t mavlink_system;
 /// @param chan		Channel to send to
 static inline bool valid_channel(mavlink_channel_t chan)
 {
+#if defined __clang__
```

Comment:
Yes gcc warns about unknown pragmas. Lots of noise! Surely  really be an issue with making clang pragmas conditional on being compiled by clang ! :)